tests/devices/qubit/test_adjoint_jacobian.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit and integration tests for the adjoint_jacobian function for DefaultQubit2"""
+import pytest
+import pennylane as qml
+from pennylane.devices.qubit import adjoint_jacobian
+from pennylane.tape import QuantumScript
+import pennylane.numpy as np
+from pennylane.devices.qubit.preprocess import validate_and_expand_adjoint
+
+
+class TestAdjointJacobian:
+    """Tests for adjoint_jacobian"""
+
+    def test_custom_wire_labels(self, tol):
+        """Test that adjoint_jacbonian works as expected when custom wire labels are used."""
+        qs = QuantumScript(
+            [qml.RX(0.123, wires="a"), qml.RY(0.456, wires="b")], [qml.expval(qml.PauliX("a"))]
+        )
+        qs.trainable_params = {0, 1}
+
+        qs_valid = validate_and_expand_adjoint(qs)
+
+        calculated_val = adjoint_jacobian(qs_valid)
+
+        tapes, fn = qml.gradients.finite_diff(qs)
+        results = tuple(qml.devices.qubit.simulate(t) for t in tapes)
```

Comment:
is this going to be the standard workflow with converting tapes and gradient functions into results with this device? cc @albi3ro

Reply:
We could create a new instance of `DefaultQubit2` and run `execute` with that. What ever gets the job done.